src/components/Home/Topbar.jsx

```diff
@@ -1,9 +1,290 @@
-import React from 'react'
+"use client";
+import Link from "next/link";
+import React, { useState } from "react";
+import { Menu, X, User, Lock, ChevronDown, AlignRight } from "react-feather";
 
 const Topbar = () => {
+  const [isMobileMenuOpen, setIsMobileMenuOpen] = useState(false);
+  const [pagesDropdownOpen, setPagesDropdownOpen] = useState(false);
+  const [speDropdownOpen, setSpeDropdownOpen] = useState(false);
+
+  const links = (
+    <>
+      <Link href="/" className="hover:text-blue-600 flex gap-1 items-center">
+        Home
+      </Link>
+
+      <div className="relative group">
+        <Link
+          href={"#"}
+          className="hover:text-blue-600 flex gap-1 items-center"
+        >
+          Pages <ChevronDown />
+        </Link>
+
+        {/* Dropdown */}
+        <div className="w-48 absolute left-0 top-6 mt-1 border rounded-sm opacity-0 group-hover:opacity-100 transition-all duration-300 ease-in-out hidden group-hover:block bg-white shadow-lg z-10">
+          <div className="bg-white w-48 py-2 opacity-0 -mt-4"></div>
+          <Link
+            href="/about"
+            className="hover:text-blue-600 flex gap-1 items-center px-4 py-3 duration-300 hover:px-6"
+          >
+            About Us
+          </Link>
+          <hr />
+          <Link
+            href="/services"
+            className="hover:text-blue-600 flex gap-1 items-center px-4 py-3 duration-300 hover:px-6"
+          >
+            Services
+          </Link>
+          <hr />
+          <Link
+            href="/serviceDetails"
+            className="hover:text-blue-600 flex gap-1 items-center px-4 py-3 duration-300 hover:px-6"
+          >
+            Service Details
```

Comment:
please reudce the code redundency, keep dropdown data in a helper component which is called UI there create a navdata.js file and import it nav data from there.